server/test/run_rule_controller_test.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:aqueduct/aqueduct.dart';
+import 'package:encrypt/encrypt.dart';
+import 'package:mock_request/mock_request.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import 'package:server/controller/run_rule_controller.dart';
+import 'package:server/proto/scheduled_rule.pb.dart';
+import 'package:server/proto/rule.pb.dart' as proto;
+import 'package:server/model/rule.dart';
+import 'package:server/service/dv360.dart';
+import 'package:server/service/firestore.dart';
+import 'package:server/service/google_api.dart';
+import 'package:server/utils.dart';
+
+class MockGoogleApi extends Mock implements GoogleApi {}
+
+class MockFirestoreClient extends Mock implements FirestoreClient {}
+
+class MockRule extends Mock implements Rule {}
+
+class TestRunRuleController extends RunRuleController {
+  TestRunRuleController(GoogleApi googleApi, FirestoreClient firestoreClient,
+      String aesKey, String baseUrl)
+      : super(googleApi, firestoreClient, aesKey, baseUrl);
+
+  @override
+  Rule getRule(proto.Rule ruleProto) {
```

Comment:
instead of overriding getRule(), can we let mockFirestoreClient.getRule() return a real rule object, then we don't have to test getRule().

Reply:
Unfortunately, that doesn't completely work because the `DisplayVideo360Client` created in the method (which I can't mock) will try to run the rule and fail. Instead, I have removed the inheritance by mocking the DV360 server.